lib/sigh/resign.rb

```diff
@@ -5,28 +5,32 @@ module Sigh
   class Resign
     def run(options, args)
       # get the command line inputs and parse those into the vars we need...
-      ipa, signing_identity, provisioning_profile = get_inputs(options, args)
+      ipa, signing_identity, provisioning_profiles = get_inputs(options, args)
 
       # ... then invoke out programmatic interface with these vars
-      resign(ipa, signing_identity, provisioning_profile)
+      resign(ipa, signing_identity, provisioning_profiles)
     end
 
-    def self.resign(ipa, signing_identity, provisioning_profile)
-      self.new.resign(ipa, signing_identity, provisioning_profile)
+    def self.resign(ipa, signing_identity, provisioning_profiles)
+      self.new.resign(ipa, signing_identity, provisioning_profiles)
     end
 
-    def resign(ipa, signing_identity, provisioning_profile)
+    def resign(ipa, signing_identity, provisioning_profiles)
       resign_path = find_resign_path
       signing_identity = find_signing_identity(signing_identity)
+      provisioning_profiles = [provisioning_profiles] unless provisioning_profiles.is_a?(Enumerable)
 
       # validate that we have valid values for all these params, we don't need to check signing_identity because `find_signing_identity` will only ever return a valid value
-      validate_params(resign_path, ipa, provisioning_profile)
+      validate_params(resign_path, ipa, provisioning_profiles)
+
+      provisioning_options = provisioning_profiles.map { |fst, snd| "-p #{[fst, snd].compact.join('=')}" }.join(' ')
```

Comment:
Line is too long. [116/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.